pulp_ansible/app/tasks/collections.py

```diff
@@ -205,3 +134,218 @@ def _update_highest_version(collection_version):
+
+                    url = response["download_url"]
+
+                    collection_version = CollectionVersion(
+                        namespace=metadata["namespace"],
+                        name=metadata["name"],
+                        version=version["version"],
+                    )
+
+                    relative_path = "%s.%s.%s" % (
+                        metadata["namespace"],
+                        metadata["name"],
+                        version["version"],
+                    )
+
+                    artifact = response["artifact"]
+
+                    d_artifact = DeclarativeArtifact(
+                        artifact=Artifact(sha256=artifact["sha256"], size=artifact["size"]),
+                        url=url,
+                        relative_path=relative_path,
+                        remote=self.remote,
+                        deferred_download=self.deferred_download,
+                    )
+                    d_content = DeclarativeContent(
+                        content=collection_version, d_artifacts=[d_artifact]
+                    )
+                    pb.increment()
+                    await self.put(d_content)
+
+    async def _fetch_collections(self):
+        async for metadata in self._fetch_galaxy_pages():
+            for result in metadata["results"]:
+
+                response = await get_json_from_url(url=result["versions_url"])
+
+                collection = {
+                    "href": result["href"],
+                    "name": result["name"],
+                    "namespace": result["namespace"]["name"],
+                    "versions": response["results"],
+                }
+                yield collection
+
+    async def _fetch_galaxy_pages(self):
+        """
+        Fetch the collections in a remote repository.
+
+        Returns:
+            async generator: dicts that represent pages from galaxy api
+
+        """
+        page_count = 0
+        remote = self.remote
+
+        with ProgressBar(message="Parsing Pages from Galaxy Collections API") as progress_bar:
+            metadata = await get_json_from_url(url=get_page_url(remote.url))
+
+            page_count = math.ceil(float(metadata["count"]) / float(PAGE_SIZE))
+            progress_bar.total = page_count
+            progress_bar.save()
+
+            yield metadata
+            progress_bar.increment()
+
+            # Concurrent downloads are limited by aiohttp...
+            not_done = set(
+                asyncio.ensure_future(get_json_from_url(url=get_page_url(remote.url, page)))
+                for page in range(2, page_count + 1)
+            )
+
+            while not_done:
+                done, not_done = await asyncio.wait(not_done, return_when=asyncio.FIRST_COMPLETED)
+                for item in done:
+                    yield item.result()
+                    progress_bar.increment()
+
+
+class AnsibleContentSaver(ContentSaver):
```

Comment:
Can this name be `CollectionContentSaver` since it never deals with Role content?